ca_on_mississauga/people.py

```diff
@@ -16,18 +16,22 @@ def scrape(self):
             if "vacant" not in councillor_url.xpath(".//div//div[1]/text()")[0].lower():
                 yield self.councillor_data(councillor_url.attrib["href"])
 
-        mayor_page = self.lxmlize(MAYOR_PAGE)
-        mayor_name = mayor_page.xpath('//*[@id="com-main"]/div/div/div/h1/text()')[0]
-        if "vacant" not in mayor_name.lower():
-            yield self.mayor_data(MAYOR_PAGE)
+        mayor_url = page.xpath('//li/a[contains(@href, "mayor")]')[0]
+        if "vacant" not in mayor_url.xpath(".//div//div[1]/text()")[0].lower():
+            yield self.mayor_data(mayor_url.attrib["href"])
 
     def councillor_data(self, url):
         page = self.lxmlize(url)
 
         name_district = page.xpath('//*[@id="com-main"]/div/div/div/h1/text()')[0]
         hyphen = name_district.find("Councillor")
+        if hyphen == -1:
+            hyphen = 9
```

Comment:
What is the significance of the magic numbers 9, 27 and 8?

Reply:
@jpmckinney so for 9 it was the length of "Ward 5 –" because unhelpfully the council didn't include the word "Councillor" on Natalie's page title https://www.mississauga.ca/council/city-council-members/ward-5-councillor-natalie-hart/ . For 27 it was the length of "Councillor and Deputy Mayor" in regards to Matt's title as i was hitting errors if the string and Deputy Mayor were not removed https://www.mississauga.ca/council/city-council-members/ward-8-councillor-matt-mahoney/ . For 8 it is the length of "Mayor - " to remove it from Carolyn's page title https://www.mississauga.ca/council/city-council-members/mayor-carolyn-parrish/ 

I wasn't sure if there were better ways to resolve these random matters

Reply:
I suggest doing things like `variable.replace("Councillor and Deputy Mayor", "").strip()`, that way if there is no match, nothing special happens. Right now, if there's no match, the string will be cut at a strange index.